index.js

```diff
@@ -219,7 +219,7 @@ var PgDriver = Base.extend({
           var searchPathes = result[0].search_path.split(',');
 
           for (var i = 0; i < searchPathes.length; ++i) {
-            if (searchPathes[i].indexOf('"') !== 0) {
+            if (searchPathes[i].indexOf('"') !== -1) {
```

Comment:
The indexOf() method returns the first index at which a given element can be found in the array, or -1 if it is not present.

So in this case shouldn't we use `searchPathes[i].indexOf('"') === -1` instead? If the search_path does not have a " we will surround with one.

Reply:
I'm not convinced this quoting code is needed _at all_.  The return from postgres of `SHOW search_path` quotes anything that needs it, so it shouldn't be necessary to apply quoting to anything that came from that, only to the new entry being added.

... I think.

Reply:
If there is no BC then there is no need for quote at all. But the bug fix that is on this PR is still not fixing anything at all, it's still duplicating the quotes.

Reply:
yes that is true, b/c it would need to be `=== -1`. However I think it might be true that `SHOW search_path` will actually not return anything that needs to be quoted. But can't guarantee it as well. Since the pg driver is also the base for pgwire drivers it could have side effects on those. Since I have no guarantee I will rather keep this to avoid a breaking change, but accept a bugfix for this of course.

Reply:
`SHOW search_path;` could return `"$user"` (with quotes) [https://www.postgresql.org/docs/current/ddl-schemas.html#:~:text=search_path%0A%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D%2D-,%22%24user%22,-%2C%20public](url). So this PR must be accepted to prevent throwing an exception.
```
[SQL] SET search_path TO "some_schema",""$user"","public"
[ERROR] AssertionError [ERR_ASSERTION]: ifError got unwanted exception: zero-length delimited identifier at or near """"
```

Reply:
we tried and merged it, but it causes the test suite to fail. this is breaking not fixing things. if anyone wants to jump on this again feel free to.